### PR TITLE
adding user authentication

### PR DESCRIPTION
**Pull request by alialsaeedi19**

### CODE REVIEW

File: api/endpointsServices/BaseService.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        else {
+          resolve(results);
+        }
+      });
+    });
+  }
+
+  update(query, update) {
+    return new Promise((resolve, reject) => {
+      this.schema.findOneAndUpdate(query, update, function (err, user) {
```
Comment (alronz) [inline api/endpointsServices/BaseService.js]: why we have "user" in the base service ?

File: api/endpointsServices/UserService.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const BaseService = require('./BaseService')
+
+class UserService extends BaseService {
+
+  constructor(userSchema) {
+    super(userSchema);
+    this.userSchema = userSchema;
```
Comment (alronz) [inline api/endpointsServices/UserService.js]: you can delete this and always use this.schema

File: api/endpointsServices/UserService.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  signIn(email, password) {
+    return new Promise((resolve, reject) => {
+      this.userSchema.findOne({
```
Comment (alronz) [inline api/endpointsServices/UserService.js]: here => this.schema is enough

File: api/endpointsServices/BaseService.js (hunk 1 above)
Comment (alialsaeedi19) [reply api/endpointsServices/BaseService.js]: fixed

**Pull request merged**